src/pages/index.js

```diff
@@ -99,53 +102,139 @@ modalWithImage.setEventListeners();
 
 // Modal Form Popup
 
-const newCardModal = new ModalWithForm(
+const addCardModal = new ModalWithForm(
   "#add-card-modal",
   handleAddCardFormSubmit
 );
 
 addNewCardButton.addEventListener("click", () => {
-  newCardModal.open();
+  addCardModal.open();
 });
 
 function handleAddCardFormSubmit(data) {
-  const name = data.title;
-  const link = data.url;
-  renderCard({ name, link }, cardListEl);
-  newCardModal.close();
+  const name = data.name;
+  const link = data.link;
+  addCardModal.setButtonText(true);
+  api
+    .newCardReq(name, link)
+    .then((result) => {
+      addCardModal.setButtonText(false);
+      const { name, link } = result;
+      renderCard(result);
+      addCardModal.close();
+      addCardFormValidator.disableButton();
+    })
+    .catch((err) => {
+      console.error(err);
+    })
+    .finally(() => {
+      saveButton.textContent = "SAVE";
```

Comment:
Please, delete it

`addCardModal.setButtonText(false);` should be in `finally`

You need to return the **default** button text only in blocks `finally` so that the button could get the text back in any case: `success` or `failure`.

**Please, correct it everywhere in the code**